### ISSUE TITLE
Bug with passing custom separator while flattening nested object

### ISSUE THREAD

**Issue by glebsts** [wnameless/json-flattener#14]
[JsonFlattener.java:264](https://github.com/wnameless/json-flattener/blob/master/src/main/java/com/github/wnameless/json/flattener/JsonFlattener.java#L264)
`return new JsonFlattener(val.toString()).withFlattenMode(flattenMode).flattenAsMap();`
is actually missing '.withSeparator(this.separator)', and currently flattens nested object with default dot separator.

Mb there is similar issue for print mode, also for unflattener

**Comment by wnameless** [comment]
You should call the .withSeparator(this.separator) method before the .flattenAsMap() or .flatten() methods.

**Comment by glebsts** [comment]
I do call it for JsonFlattener instance I create in y code. But this instance internally creates another one at the row I linked, and does not give it its separator.

**Comment by wnameless** [comment]
Of course the separator will be lost. JsonFlattener doesn't provide any global configuration. Every new JsonFlattener instance resets all configuration to the its default.

If you are going to return a new Map with self-defined separators, you should do as the following line:

return new JsonFlattener(val.toString()).withSeparator(this.separator).withFlattenMode(flattenMode).flattenAsMap();

or

If you already assigned the internal JsonFlattener to a variable:
JsonFlattener jf = JsonFlattener(val.toString()).withSeparator(this.separator).withFlattenMode(flattenMode);

you can simply use this one instead:
return  jf.flattenAsMap();

**Comment by glebsts** [comment]
But this instance is created in JsonFlattener itself, which is supposed to be working dependency library :) I think we do not understand each other. I made a test which is a copy of existing separator test, but with existing test5.json payload. Do you see the difference in flattening 'i'?

```
  @Test
  public void testWithSeparatorAndNestedObject() throws IOException {
    URL url = Resources.getResource("test5.json");
    String json = Resources.toString(url, Charsets.UTF_8);
    assertEquals(
        "{\"a_b\":1,\"a_c\":null,\"a_d\":[false,{\"i_j\":[false,true]}],\"e\":\"f\",\"g\":2.3,\"z\":{}}",
        new JsonFlattener(json).withFlattenMode(FlattenMode.KEEP_ARRAYS)
            .withSeparator('_')
            .flatten());
  }
```


```
org.junit.ComparisonFailure: 
Expected :{"a_b":1,"a_c":null,"a_d":[false,{"i_j":[false,true]}],"e":"f","g":2.3,"z":{}}
Actual   :{"a_b":1,"a_c":null,"a_d":[false,{"i.j":[false,true]}],"e":"f","g":2.3,"z":{}}
```

**Comment by wnameless** [comment]
The issue has been fixed in version 0.2.3. Sorry for the late fix.
